cmd/test.go

```diff
@@ -331,6 +331,40 @@ func (t *Test) GetCmd() *cobra.Command {
 				}
 			}
 
+			mongoPassword, err := cmd.Flags().GetString("mongoPassword")
+			if err != nil {
+				t.logger.Error("failed to read the mongo password")
+				return err
+			}
+
+			t.logger.Debug("the configuration for mocking mongo connection", zap.Any("password", mongoPassword))
+			//Check if app command starts with docker or  docker-compose.
+			dockerRelatedCmd, dockerCmd := utils.IsDockerRelatedCmd(appCmd)
+			if !isDockerCmd && dockerRelatedCmd {
+				isDockerCompose := false
+				if dockerCmd == "docker-compose" {
+					isDockerCompose = true
+				}
+				testCfg := utils.TestFlags{
+					Path:               path,
+					Proxyport:          proxyPort,
+					Command:            appCmd,
+					Testsets:           testsets,
+					ContainerName:      appContainer,
+					NetworkName:        networkName,
+					Delay:              delay,
+					BuildDelay:         buildDelay,
+					ApiTimeout:         apiTimeout,
+					PassThroughPorts:   ports,
+					ConfigPath:         configPath,
+					MongoPassword:      mongoPassword,
+					CoverageReportPath: coverageReportPath,
+					EnableTele:         enableTele,
+				}
```

Comment:
You are not setting `WithCoverage` field of `TestFlags` struct

Reply:
done